src/libs/events.ts

```diff
@@ -4,6 +4,8 @@ import { Ajax } from 'src/libs/ajax';
 import { useRoute } from 'src/libs/nav';
 import { containsProtectedDataPolicy, WorkspaceInfo, WorkspaceWrapper } from 'src/libs/workspace-utils';
 
+type NestedEnum<T> = { [key: string]: T | NestedEnum<T> };
```

Comment:
I think this is more of a NestedKeySet... enums have keyName: 'keyName' (name === value), and these do not.

Reply:
Oh wait, it ends up "NestedKeyValues" / "NestedKeyValuesValue"...

Reply:
Thinking about it some more... this type isn't exported, so there's no real need to come up with a fitting generic name.

Reply:
Updated in https://github.com/DataBiosphere/terra-ui/pull/4308/commits/0070dc5dca7ff1c6185d0cba78c4f702539b02da. Let me know what you think.